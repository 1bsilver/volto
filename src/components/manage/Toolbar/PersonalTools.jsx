import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import jwtDecode from 'jwt-decode';

import { Icon } from '../../../components';

import { getUser } from '../../../actions';
import logoutSVG from '../../../icons/log-out.svg';
import rightArrowSVG from '../../../icons/right-key.svg';
import avatar from './avatar.jpg';
import backSVG from '../../../icons/back.svg';

/**
 * Toolbar container class.
 * @class PersonalTools
 * @extends Component
 */
@connect(
  state => ({
    user: state.users.user,
    loaded: state.users.get.loaded,
    loading: state.users.update.loading,
    userId: state.userSession.token
      ? jwtDecode(state.userSession.token).sub
      : '',
  }),
  { getUser },
)
class PersonalTools extends Component {
  /**
   * Property types.
   * @property {Object} propTypes Property types.
   * @static
   */
  static propTypes = {
    user: PropTypes.shape({
      fullname: PropTypes.string,
      email: PropTypes.string,
      home_page: PropTypes.string,
      location: PropTypes.string,
    }).isRequired,
    userId: PropTypes.string.isRequired,
    getUser: PropTypes.func.isRequired,
    loaded: PropTypes.bool.isRequired,
    loading: PropTypes.bool,
    loadComponent: PropTypes.func.isRequired,
    componentIndex: PropTypes.number.isRequired,
  };

  /**
   * Component will mount
   * @method componentWillMount
   * @returns {undefined}
   */
  componentDidMount() {
    this.props.getUser(this.props.userId);
  }

  push = selector => {
    this.setState(() => ({
      pushed: true,
    }));
    this.props.loadComponent(selector);
  };

  pull = () => {
    this.props.unloadComponent();
  };

  /**
   * Render method.
   * @method render
   * @returns {string} Markup for the component.
   */
  render() {
    return (
      <div
        className="personal-tools pastanaga-menu"
        style={{
          left: `${this.props.componentIndex * 100}%`,
        }}
      >
        <header className="header">
          <button className="back" onClick={this.pull}>
            <Icon name={backSVG} size="32px" />
          </button>
          <div className="vertical divider" />
          <h2>
            {this.props.user.fullname
              ? this.props.user.fullname
              : this.props.user.username}
          </h2>
          <Link to="/logout">
            <Icon className="logout" name={logoutSVG} size="32px" />
          </Link>
        </header>
        <div className="avatar">
          <img src={avatar} alt="user avatar" />
        </div>
        <div className="stats">
          {/* This should be a Component by itself*/}
          <ul>
            <li>
              <span>126</span>
              <span>Items Created</span>
            </li>
            <li>
              <span>43</span>
              <span>Uploads</span>
            </li>
            <li>
              <span>13</span>
              <span>Reviews</span>
            </li>
          </ul>
        </div>
        <div className="pastanaga-menu-list">
          {/* This (probably also) should be a Component by itself*/}
          <ul>
            <li>
              <button onClick={() => this.push('profile')}>
                Profile
                <Icon name={rightArrowSVG} size="24px" />
              </button>
            </li>
            <li>
              <Link to="/personal-preferences">
                Preferences
                <Icon name={rightArrowSVG} size="24px" />
              </Link>
            </li>
            <li>
              <Link to="/controlpanel">
                Site Setup
                <Icon name={rightArrowSVG} size="24px" />
              </Link>
            </li>
          </ul>
        </div>
      </div>
    );
  }
}

export default PersonalTools;
