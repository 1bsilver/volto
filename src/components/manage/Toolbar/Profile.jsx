import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Icon } from '../../../components';
import backSVG from '../../../icons/back.svg';

/**
 * Profile container class.
 * @class Profile
 * @extends Component
 */
class Profile extends Component {
  static propTypes = {
    unloadComponent: PropTypes.func.isRequired,
    componentIndex: PropTypes.number.isRequired,
  };

  pull = () => {
    this.props.unloadComponent();
  };

  /**
   * Render method.
   * @method render
   * @returns {string} Markup for the component.
   */
  render() {
    return (
      <div
        className="profile pastanaga-menu"
        style={{
          width: '100vw',
        }}
      >
        <header className="header pulled">
          <button onClick={this.pull}>
            <Icon name={backSVG} size="32px" />
          </button>
          <div className="vertical divider" />
          <h2>Profile</h2>
        </header>
      </div>
    );
  }
}

export default Profile;
