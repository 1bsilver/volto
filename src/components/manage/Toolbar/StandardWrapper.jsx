import React from 'react';
import PropTypes from 'prop-types';
import cx from 'classnames';
import { Icon } from '../../../components';
import backSVG from '../../../icons/back.svg';

const StandardWrapper = props => {
  function pull() {
    props.unloadComponent();
  }

  return (
    <div
      className={cx(`${props.componentName} pastanaga-menu`, {
        'has-inner-actions': props.hasActions,
      })}
    >
      <header className="header pulled">
        <button onClick={pull}>
          <Icon name={backSVG} size="32px" />
        </button>
        <div className="vertical divider" />
        <h2>{props.componentName}</h2>
      </header>
      {props.children}
    </div>
  );
};

StandardWrapper.propTypes = {
  componentName: PropTypes.string.isRequired,
  unloadComponent: PropTypes.func.isRequired,
  loadComponent: PropTypes.string.isRequired,
  closeMenu: PropTypes.string.isRequired,
  hasActions: PropTypes.bool,
};

export default StandardWrapper;
