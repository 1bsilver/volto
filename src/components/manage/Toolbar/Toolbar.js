/**
 * Toolbar component.
 * @module components/Toolbar
 */

import React from 'react';

/**
 * Toolbar component class.
 * @function Toolbar
 * @returns {string} Markup of the component.
 */
const Toolbar = () =>
  <div id="edit-zone" role="toolbar" className="pat-toolbar initialized">
    <div className="plone-toolbar-container">
      <nav>
        <ul className="plone-toolbar-main" >
          <li>
            <a href="/view">
              <span aria-hidden="true" className="icon-view" />
              <span>View</span>
            </a>
          </li>
          <li>
            <a href="/edit">
              <span aria-hidden="true" className="icon-edit" />
              <span>Edit</span>
            </a>
          </li>
        </ul>
      </nav>
    </div>
  </div>;

export default Toolbar;
