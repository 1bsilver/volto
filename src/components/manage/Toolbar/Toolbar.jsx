/**
 * Toolbar component.
 * @module components/manage/Toolbar/Toolbar
 */

import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { Link } from 'react-router';
import { Dropdown, Icon, Menu } from 'semantic-ui-react';
import jwtDecode from 'jwt-decode';

import { Actions, Display, Types, Workflow } from '../../../components';

import logo from './plone-toolbarlogo.svg';

/**
 * Toolbar container class.
 * @class Toolbar
 * @extends Component
 */
@connect(state => ({
  token: state.userSession.token,
  fullname: jwtDecode(state.userSession.token).fullname,
}))
export default class Toolbar extends Component {
  /**
   * Property types.
   * @property {Object} propTypes Property types.
   * @static
   */
  static propTypes = {
    pathname: PropTypes.string.isRequired,
    selected: PropTypes.string.isRequired,
    token: PropTypes.string,
    fullname: PropTypes.string,
  };

  /**
   * Default properties.
   * @property {Object} defaultProps Default properties.
   * @static
   */
  static defaultProps = {
    token: null,
    fullname: '',
  };

  /**
   * Render method.
   * @method render
   * @returns {string} Markup for the component.
   */
  render() {
    return (
      this.props.token &&
      <Menu inverted vertical fixed="left">
        <Menu.Item color="blue" active>
          <img alt="Plone Toolbar" src={logo} />
        </Menu.Item>
        <Link
          to={`${this.props.pathname}/edit`}
          className={`item${this.props.selected === 'edit' ? ' active' : ''}`}
        >
          <span><Icon name="write" /> Edit</span>
        </Link>
        <Link
          to={this.props.pathname}
          className={`item${this.props.selected === 'view' ? ' active' : ''}`}
        >
          <span><Icon name="eye" /> View</span>
        </Link>
        <Types
          pathname={this.props.pathname}
          active={this.props.selected === 'add'}
        />
        <Workflow pathname={this.props.pathname} />
        <Actions pathname={this.props.pathname} />
        <Display pathname={this.props.pathname} />
        <Link
          to={`${this.props.pathname}/sharing`}
          className={`item${this.props.selected === 'sharing' ? ' active' : ''}`}
        >
          <span><Icon name="users" /> Sharing</span>
        </Link>
        <Dropdown
          className="personal-bar"
          item
          trigger={<span><Icon name="user" /> {this.props.fullname}</span>}
          pointing="left"
        >
          <Dropdown.Menu>
            <Link to="/logout" className="item">
              <span><Icon name="sign out" /> Log out</span>
            </Link>
          </Dropdown.Menu>
        </Dropdown>
      </Menu>
    );
  }
}
