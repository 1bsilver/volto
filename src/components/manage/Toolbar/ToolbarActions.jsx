import React from 'react';
import { filter, find } from 'lodash';
import { Link } from 'react-router-dom';
import { defineMessages, injectIntl } from 'react-intl';
import penSVG from '@plone/volto/icons/pen.svg';
import { Icon } from '@plone/volto/components/theme/Icon/Icon';
import { getBaseUrl } from '@plone/volto/helpers';

const messages = defineMessages({
  edit: {
    id: 'Edit',
    defaultMessage: 'Edit',
  },
  contents: {
    id: 'Contents',
    defaultMessage: 'Contents',
  },
  add: {
    id: 'Add',
    defaultMessage: 'Add',
  },
  more: {
    id: 'More',
    defaultMessage: 'More',
  },
  personalTools: {
    id: 'Personal tools',
    defaultMessage: 'Personal tools',
  },
  shrinkToolbar: {
    id: 'Shrink toolbar',
    defaultMessage: 'Shrink toolbar',
  },
  personalInformation: {
    id: 'Personal Information',
    defaultMessage: 'Personal Information',
  },
  personalPreferences: {
    id: 'Personal Preferences',
    defaultMessage: 'Personal Preferences',
  },
  collection: {
    id: 'Collection',
    defaultMessage: 'Collection',
  },
  file: {
    id: 'File',
    defaultMessage: 'File',
  },
  link: {
    id: 'Link',
    defaultMessage: 'Link',
  },
  newsItem: {
    id: 'News Item',
    defaultMessage: 'News Item',
  },
  page: {
    id: 'Page',
    defaultMessage: 'Page',
  },
  back: {
    id: 'Back',
    defaultMessage: 'Back',
  },
});

const EditAction = (props) => {
  const editAction = find(this.props.actions.object, { id: 'edit' });
  const path = getBaseUrl(this.props.pathname);

  return (
    editAction && (
      <Link
        aria-label={this.props.intl.formatMessage(messages.edit)}
        className="edit"
        to={`${path}/edit`}
      >
        <Icon name={penSVG} size="30px" className="circled" />
      </Link>
    )
  );
};
