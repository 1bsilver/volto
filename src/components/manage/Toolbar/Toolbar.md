Toolbar example source:

```jsx_ noeditor
import Toolbar from './Toolbar';
import {StaticRouter} from 'react-router-dom';

const { Provider } = require('react-intl-redux');
const configureStore = require('redux-mock-store').default;

const store = configureStore()({
  userSession: {
    login: {},
  },
  intl: {
    locale: 'en',
    messages: {},
  },
  users: {
    reset: {
      loading: '',
    },
  },
  actions: {},
  content: { get: {}, data: {} },
  controlpanels: {},
  diff: {},
  history: [],
  schema: {},
});

<div className={'rsg--pre-42'}>
  <Provider store={store}>
    <StaticRouter>
      <Toolbar
        token=""
        content={{
          '@type': '',
          is_folderish: true,
          review_state: 'open',
        }}
        inner={{}}
      />
    </StaticRouter>
  </Provider>
</div>;
```
