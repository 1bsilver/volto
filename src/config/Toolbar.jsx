import More from '@plone/volto/components/manage/Toolbar/More';
import PersonalTools from '@plone/volto/components/manage/Toolbar/PersonalTools';
import Types from '@plone/volto/components/manage/Toolbar/Types';
import Profile from '@plone/volto/components/manage/Toolbar/Profile';

export const defaultToolbarComponents = {
  personalTools: PersonalTools,
  more: More,
  types: Types,
  profile: Profile,
};
