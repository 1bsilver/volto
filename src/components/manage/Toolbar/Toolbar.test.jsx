import React from 'react';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-intl-redux';
import jwt from 'jsonwebtoken';

import Toolbar from './Toolbar';

const mockStore = configureStore();

jest.mock('../Actions/Actions', () => jest.fn(() => <div id="actions" />));
jest.mock('../Display/Display', () => jest.fn(() => <div id="display" />));
jest.mock('../Types/Types', () => jest.fn(() => <div id="types" />));
jest.mock('../Workflow/Workflow', () => jest.fn(() => <div id="workflow" />));

describe('Toolbar', () => {
  it('renders an actions component', () => {
    const store = mockStore({
      userSession: {
        token: jwt.sign({ fullname: 'John Doe' }, 'secret'),
      },
      content: {
        data: {
          '@type': 'Folder',
        },
      },
      intl: {
        locale: 'en',
        messages: {},
      },
    });
    const component = renderer.create(
      <Provider store={store}>
        <Toolbar pathname="/test" selected="view" />
      </Provider>,
    );
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });
});
