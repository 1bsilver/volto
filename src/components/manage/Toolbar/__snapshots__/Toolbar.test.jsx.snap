// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Toolbar renders an actions component 1`] = `
<div
  className="ui inverted vertical left fixed menu"
>
  <div
    className="blue active item"
    onClick={[Function]}
  >
    <img
      alt="Plone Toolbar"
      src="plone-toolbarlogo.svg"
    />
  </div>
  <a
    className="item"
    onClick={[Function]}
    style={Object {}}
  >
    <span>
      <i
        aria-hidden="true"
        className="write icon"
      />
       Edit
    </span>
  </a>
  <a
    className="item active"
    onClick={[Function]}
    style={Object {}}
  >
    <span>
      <i
        aria-hidden="true"
        className="eye icon"
      />
       View
    </span>
  </a>
  <div
    id="types"
  />
  <div
    id="workflow"
  />
  <div
    id="actions"
  />
  <div
    id="display"
  />
  <a
    className="item"
    onClick={[Function]}
    style={Object {}}
  >
    <span>
      <i
        aria-hidden="true"
        className="users icon"
      />
       Sharing
    </span>
  </a>
  <div
    aria-busy={undefined}
    aria-disabled={undefined}
    aria-expanded={false}
    aria-multiselectable={undefined}
    className="ui item left pointing personal-bar dropdown"
    onBlur={[Function]}
    onChange={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onMouseDown={[Function]}
    role="listbox"
    tabIndex={0}
  >
    <span>
      <i
        aria-hidden="true"
        className="user icon"
      />
       
      John Doe
    </span>
    <i
      aria-hidden="true"
      className="dropdown icon"
    />
    <div
      className="menu transition"
    >
      <a
        className="item"
        onClick={[Function]}
        style={Object {}}
      >
        <span>
          <i
            aria-hidden="true"
            className="sign out icon"
          />
           Log out
        </span>
      </a>
    </div>
  </div>
</div>
`;
